model/nc-news.model.js

```diff
@@ -17,30 +17,64 @@ function fetchArticleById(article_id) {
     });
 }
 
-function fetchAllArticles() {
-  return db
-    .query(
-      `SELECT 
-        articles.article_id, 
-        articles.title, 
-        articles.topic, 
-        articles.author, 
-        articles.created_at, 
-        articles.votes, 
-        articles.article_img_url,
-        COUNT(comments.comment_id)::INT AS comment_count
-      FROM articles
-      LEFT JOIN comments 
-      ON articles.article_id = comments.article_id
-      GROUP BY articles.article_id
-      ORDER BY articles.created_at DESC;`
-    )
-    .then(({ rows }) => {
-      return rows;
-    })
-    .catch((err) => {
-      throw err;
-    });
+function fetchAllArticles(sort_by = "created_at", order = "desc", topic) {
+  const allowedSortByColumns = [
+    "created_at",
+    "title",
+    "votes",
+    "author",
+    "topic",
+    "article_id",
+    "comment_count",
+  ];
+  const allowedOrder = ["asc", "desc"];
+
+  if (!allowedSortByColumns.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "Invalid sort by column" });
+  }
+  if (!allowedOrder.includes(order.toLowerCase())) {
+    return Promise.reject({ status: 400, msg: "Invalid order" });
+  }
+
+  let queryStr = `
+    SELECT 
+      articles.article_id, 
+      articles.title, 
+      articles.topic, 
+      articles.author, 
+      articles.created_at, 
+      articles.votes, 
+      articles.article_img_url,
+      COUNT(comments.comment_id)::INT AS comment_count
+    FROM articles
+    LEFT JOIN comments 
+    ON articles.article_id = comments.article_id
+  `;
+  const queryValues = [];
+
+  if (topic) {
+    queryStr += " WHERE articles.topic = $1";
+    queryValues.push(topic);
+  }
+
+  queryStr += `
+    GROUP BY articles.article_id
+    ORDER BY ${sort_by} ${order.toUpperCase()}
+  `;
```

Comment:
This model works as you currently have it set up, but I think you can tidy up the code below if you were to turn this part into an if/else. Have a think about you could achieve this, as it'll make the code a bit cleaner 👍